From ef7253365c420ca486f074b9e9119a222e30fea0 Mon Sep 17 00:00:00 2001
From: Afzal Patel <122491982+afzpatel@users.noreply.github.com>
Date: Mon, 9 Mar 2026 12:21:36 -0400
Subject: [PATCH] [rocprofiler-sdk][rocprofiler-register] add CPackComponent
 (#3499)

* [rocprofiler-sdk][rocprofiler-regsiter] add CPackComponent

* add includes to the top
---
 .../cmake/rocprofiler_register_config_packaging.cmake           | 1 +
 .../rocprofiler-sdk/cmake/rocprofiler_config_packaging.cmake    | 2 ++
 2 files changed, 3 insertions(+)

diff --git a/projects/rocprofiler-register/cmake/rocprofiler_register_config_packaging.cmake b/projects/rocprofiler-register/cmake/rocprofiler_register_config_packaging.cmake
index bb7a96c1d35479d8d2e077b9b499f72ca27bc8ac..e1f7b069420e4662c7a39d35bc5ca23db4f42c2b 100644
--- a/projects/rocprofiler-register/cmake/rocprofiler_register_config_packaging.cmake
+++ b/projects/rocprofiler-register/cmake/rocprofiler_register_config_packaging.cmake
@@ -1,4 +1,5 @@
 # configure packaging
+include(CPackComponent)
 
 # Add packaging directives
 set(CPACK_PACKAGE_NAME ${PROJECT_NAME})
diff --git a/projects/rocprofiler-sdk/cmake/rocprofiler_config_packaging.cmake b/projects/rocprofiler-sdk/cmake/rocprofiler_config_packaging.cmake
index 221abf49b68158a3f9bf4bcff2e54cc86120611f..ab412c40486504e47484fab142591f68cc70aa35 100644
--- a/projects/rocprofiler-sdk/cmake/rocprofiler_config_packaging.cmake
+++ b/projects/rocprofiler-sdk/cmake/rocprofiler_config_packaging.cmake
@@ -2,6 +2,8 @@
 # configure packaging settings
 #
 
+include(CPackComponent)
+
 function(rocprofiler_set_package_depends _VARIABLE _VALUE _INFO _REPLACE_PARENTHESIS)
     string(REPLACE ";" ", " _DEPENDS "${_VALUE}")
     if(_REPLACE_PARENTHESIS)
